core-api/src/main/java/com/optimizely/ab/event/internal/EventBuilder.java

```diff
@@ -61,21 +67,77 @@ public EventBuilder(EventBatch.ClientEngine clientEngine, String clientVersion)
         this.clientEngine = clientEngine;
         this.clientVersion = clientVersion;
         this.serializer = DefaultJsonSerializer.getInstance();
+
+        Callable<EventBatch> callable = new Callable<EventBatch>() {
+            @Override
+            public EventBatch call() throws Exception {
+                EventBatch eventBatch = null;
+                try {
+                    eventBatch = primeSerializer();
+                } catch (Exception e) {
+                    logger.error("Problem priming the JsonSerializer ", e);
+                }
+                finally {
+                    return eventBatch;
+                }
+            }
+        };
+
+        ExecutorService ex = Executors.newSingleThreadExecutor();
+        primedEvent = ex.submit(callable);
+        ex.shutdown();
+
+    }
+
+    /**
+     * primeSerializer primes the JsonSerializer and Json annotations from the EventBatch object and it's contained
+     * hierarchy.  The introspection overhead is very high for the first time hit.  This sets up all the serializer
+     * caching, taking the burden off of the first time hit in sending an event.
+     */
+    private EventBatch primeSerializer() {
+        Decision decision = new Decision("", "",
+                "", false);
+        Event impressionEvent = new Event(System.currentTimeMillis(),UUID.randomUUID().toString(), "",
+                ACTIVATE_EVENT_KEY, null, null, null, ACTIVATE_EVENT_KEY, null);
+
+        Snapshot snapshot = new Snapshot(Arrays.asList(decision), Arrays.asList(impressionEvent));
+        Visitor visitor = new Visitor("", null, new ArrayList<Attribute>(), Arrays.asList(snapshot));
+        List<Visitor> visitors = Arrays.asList(visitor);
+        EventBatch eventBatch = new EventBatch(clientEngine.getClientEngineValue(), clientVersion, "", visitors, true, "", "");
+        String payload = this.serializer.serialize(eventBatch);
+        logger.debug("JSON Serializer primed for payload {}", payload);
+
+        return  eventBatch;
     }
 
+    private void waitForPrimedEvent() {
+        if (primedEvent != null) {
+            logger.debug("had to wait for prime");
+            try {
+                primedEvent.get();
+            }
+            catch (Exception e) {
+                logger.error("Problem getting lock ", e);
```

Comment:
Can you give a bit context in this message so the user will have an easier time figuring what's wrong?